refactor (material/core): create new per-component focus indicator customization styles in the component-level stylesheets for the old and MDC library.

jackierchu opened the pull request:
Per step 2 of HCM Focus Indicator DD (http://shortn/_QfjugJram1), this PR will:

1. Create and add new per-component customizations for all appropriate stylesheets within every component
2. Utilized the newly created focus indicator width variables (mat-focus-indicator-border-width and default-focus-indicator-width) for the per-component customizations (such as margin) as configured by the mixin. 
3. This PR no longer uses the $border-width variable and config object. Instead, this PR uses the CSS variables introduced in above for customizations. Therefore, the per-component customizations no longer depend on $border-width. Rather, it depends on its equivalent CSS variable for $border-width that was created in step 1 of the HCM Focus Indicator DD.

zelliott commented on src/material-experimental/mdc-button/button.scss:
```diff
@@ -106,3 +106,26 @@
   border-width: $offset;
 }
+
+// By default, all focus indicators are flush with the bounding box of their
+// host element. For the button element, default inset/offset
```
I don't think we need to repeat the "By default, all focus indicators are flush with the bounding box of their host element." blurb across all components, I'd vote to remove it.

zelliott commented on src/material-experimental/mdc-button/button.scss:
```diff
@@ -106,3 +106,26 @@
+  margin: calc(-(var(--mat-mdc-focus-indicator-border-width, $default-focus-indicator-width) + 3px));
+}
+
+// Render the focus indicator on focus. Defining a pseudo element's
+// content will cause it to render.
```
I don't think we need to repeat this comment across all components either.

zelliott commented on src/material/button/button.scss:
```diff
@@ -153,6 +153,22 @@
+}
+
+.mat-focus-indicator.mat-stroked-button::before
+{
```
This formatting doesn't look quite right, I'd expect this `{` to be on the same line above.

zelliott reviewed (commented):
I think this looks really good! The diff is a little off - it looks like it includes changes from https://github.com/angular/components/pull/24524, it'd be great to get that fixed.

Also - at this point I think we should be able to run the Angular dev demo app and see focus indicators working (maybe)? There should still be double focus indicators in HCM, but it might be a good idea to run the dev demo app and see what the actual state of things are.

zelliott commented on src/material-experimental/mdc-chips/chip.scss:
```diff
@@ -206,3 +206,24 @@
+// In the chips the individual actions have focus so we target a different element.
+.mat-mdc-chip-action:focus .mat-mdc-focus-indicator::before {
+  content: '';
+}
```
nit: Missing newline at EOF

zelliott reviewed (approved):
LGTM!

jackierchu commented:
Took a look at the `angular:focus-indicators` branch and noticed the branch was missing the changes from my previous PR: https://github.com/angular/components/pull/24524 (which created new structural focus indicator styles, updated core stylesheet, removed theme stylesheets, etc). Although that PR was merged into the `angular:focus-indicators` branch, a force push occurred here in this PR (possibly without fetching from the angular/focus-indicators branch and its changes beforehand) which overrode the changes from my previous PR with the master branch. In turn, this is causing the merge conflicts here in this PR. Is someone with write-access able to recover changes from the previous PR (https://github.com/angular/components/pull/24524) and revert the changes back into the `angular:focus-indicators` branch? Thanks so much for the help and time!

crisbeto reviewed (approved):
LGTM
